### DISCUSSION TITLE
[feature] API to add new verified domains

### DISCUSSION THREAD

**Discussion by antoinedc** [reown-com/appkit#3699] [Ideas]
### What problem does this new feature solve?

I'm building a block explorer as a service (https://tryethernal.com), and customers can host their explorer on their own subdomain.
This explorer includes a contract interaction part, where users can send transactions through the UI.
At the moment, the only way to connect a wallet is through Metamask.

I can't switch to WC because I need to manually whitelist every domain, and this would slow down the process of setting up an explorer considerably if I need to do that when a customer adds a new domain.

An alternative would be to ask the customer to create WC account and copy/paste the verification code, but that isn't an ideal UX.

### Describe the solution you'd like

Ideally, I'd like to have an API that I can send a new domain to, and that returns the verification code.
On my end, I'll be able to generate a dynamic /.well-known/walletconnect.txt file with all the verification codes.

**Comment by rtomas** [comment]
Hi @antoinedc,
The team really like the idea and we are definitely on plans to do it. 
I don't think it will be available in the near future.

**Comment by stedyyulius** [comment]
any update on this?

**Reply by riveign** [reply]
Hi @stedyyulius nice to meet you! We have this available currently but only as a paid feature, is this something you would be willing to explore? Let me know and we can schedule a call and give you more details!
